assets/js/components/ClusterDetails/ClusterDetailsPage.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+
+import {
+  getCluster,
+  getClusterHosts,
+  getClusterHostIDs,
+} from '@state/selectors/cluster';
+import {
+  updateLastExecution,
+  executionRequested,
+} from '@state/actions/lastExecutions';
+import { getLastExecution } from '@state/selectors/lastExecutions';
+import ClusterDetails from './ClusterDetails';
+import { getClusterName } from '../ClusterLink';
+
+export function ClusterDetailsPage() {
+  const { clusterID } = useParams();
+  const navigate = useNavigate();
+
+  const cluster = useSelector(getCluster(clusterID));
+
+  const dispatch = useDispatch();
+  const lastExecution = useSelector(getLastExecution(clusterID));
+  const hosts = useSelector(getClusterHostIDs(clusterID));
+  useEffect(() => {
+    dispatch(updateLastExecution(clusterID));
+  }, [dispatch]);
+
+  const clusterHosts = useSelector(getClusterHosts(clusterID));
+
+  if (!cluster) {
+    return <div>Loading...</div>;
+  }
+
+  const renderedNodes = cluster.details?.nodes?.map((node) => ({
```

Comment:
This is the major change in the refactoring. Beforehand, this conversion was handle with much more code.
I think it is enough to "merge" the hosts information with the cluster node details